onnxruntime/core/providers/shared/node_unit.cc

```diff
@@ -0,0 +1,144 @@
+
+  const std::vector<const Node*> GetAllNodes() const noexcept override { return all_nodes_; }
+
+  INodeUnit::Type UnitType() const noexcept override { return INodeUnit::Type::Node; }
+
+ private:
+  const Node& node_;
+  std::vector<const Node*> all_nodes_;
+};
+
+class QDQNodeUnit : public INodeUnit {
+ public:
+  QDQNodeUnit(const GraphViewer& graph_viewer, const QDQ::NodeGroup& qdq_group);
+
+  const ConstPointerContainer<std::vector<NodeArg*>> InputDefs() const noexcept override {
+    return ConstPointerContainer<std::vector<NodeArg*>>(input_defs_);
+  }
+
+  const ConstPointerContainer<std::vector<NodeArg*>> OutputDefs() const noexcept override {
+    return ConstPointerContainer<std::vector<NodeArg*>>(output_defs_);
+  }
+
+  const std::string& OpType() const noexcept override { return node_.OpType(); }
+  int SinceVersion() const noexcept override { return node_.SinceVersion(); }
+  const std::string& Domain() const noexcept override { return node_.Domain(); }
+  const Path& ModelPath() const noexcept override { return node_.ModelPath(); }
+  const std::string& Name() const noexcept override { return node_.Name(); }
+
+  const Node& GetNode() const noexcept override { return node_; }
+  NodeIndex Index() const noexcept override { return node_.Index(); }
+  ProviderType GetExecutionProviderType() const noexcept override { return node_.GetExecutionProviderType(); }
+
+  const std::vector<const Node*> GetAllNodes() const noexcept override { return all_nodes_; }
+
+  INodeUnit::Type UnitType() const noexcept override { return INodeUnit::Type::QDQ; }
+
+ private:
+  void init();
+  const GraphViewer& graph_viewer_;
+  const QDQ::NodeGroup qdq_group_;
+  const Node& node_;
+  std::vector<NodeArg*> input_defs_;
+  std::vector<NodeArg*> output_defs_;
+  std::vector<const Node*> all_nodes_;
+};
+
+// QUESTION/TODO, do we want to embed the graph_viewer into the QDQNodeGroup?
+QDQNodeUnit::QDQNodeUnit(const GraphViewer& graph_viewer, const QDQ::NodeGroup& qdq_group)
+    : graph_viewer_(graph_viewer),
+      qdq_group_(qdq_group),
+      node_(*graph_viewer_.GetNode(qdq_group_.target_node)) {
+  init();
+}
+
+void QDQNodeUnit::init() {
+  for (auto node_index : qdq_group_.dq_nodes) {
+    all_nodes_.push_back(graph_viewer_.GetNode(node_index));
+  }
+
+  for (auto node_index : qdq_group_.q_nodes) {
+    auto& node = const_cast<Node&>(*graph_viewer_.GetNode(node_index));
+    for (auto* def : node.MutableOutputDefs()) {
+      output_defs_.push_back(def);
+    }
+    all_nodes_.push_back(&node);
+  }
+
+  all_nodes_.push_back(&node_);
+
+  auto get_input = [&](std::vector<NodeArg*>& defs, NodeIndex node_index) {
+    // This is a bit hacky, but seems there is not other way to get a non-const NodeArg* from a const Node
+    auto& node = const_cast<Node&>(*graph_viewer_.GetNode(node_index));
+    defs.push_back(node.MutableInputDefs()[0]);
+  };
+
```

Comment:
Significant design smell if we need const_cast to make it work. 

Instead of trying to force a QDQ node group into the existing onnxruntime::Node interface, can we instead determine what specific information we need for the support checker and create a simple class with an interface that provides just that? Instances of that class would be created from either a Node or a QDQ node group, and the support checker logic would change to use that class instead of the current Node based implementation. Pretty similar we'd have the freedom to keep that interface as clean and simple as possible. 

This approach would hopefully mean you could handle the case where there's no equivalent QLinear* operator more easily. The inputs would be in the usual order. If they're coming from a DQ node you'd have optional metadata for zp and scale and the edge/s from the DQ node. If the input is not coming from a DQ node it would be the original edge info and the optional metadata wouldn't be populated.